lib/components/src/tooltip/TooltipLinkList.tsx

```diff
@@ -50,16 +53,88 @@ const Item: FunctionComponent<TooltipLinkListProps['links'][number]> = (props) =
   );
 };
 
+function nextItem(list: HTMLUListElement, item: HTMLElement): HTMLElement {
+  if (list === item) {
+    return list.firstChild as HTMLElement;
+  }
+  if (item && item.nextElementSibling) {
+    return item.nextElementSibling as HTMLElement;
+  }
+  return list.firstChild as HTMLElement;
+}
+
+function previousItem(list: HTMLUListElement, item: HTMLElement): HTMLElement {
+  if (list === item) {
+    return list.lastChild as HTMLElement;
+  }
+  if (item && item.previousElementSibling) {
+    return item.previousElementSibling as HTMLElement;
+  }
+  return list.lastChild as HTMLElement;
+}
+
+function moveFocus(
+  list: HTMLUListElement,
+  currentFocus: HTMLElement,
+  traversalFunction: (list: HTMLUListElement, item: Element) => HTMLElement | null
+) {
+  let wrappedOnce = false;
+  const nextFocus = traversalFunction(list, currentFocus);
+
+  while (nextFocus) {
+    // Prevent infinite loop.
+    if (nextFocus === list.firstChild) {
+      if (wrappedOnce) {
+        return;
+      }
+      wrappedOnce = true;
+    }
+
+    nextFocus.focus();
+    return;
+  }
+}
+
 export const TooltipLinkList: FunctionComponent<TooltipLinkListProps> = ({
   links,
   LinkWrapper,
-}) => (
-  <List>
-    {links.map(({ isGatsby, ...p }) => (
-      <Item key={p.id} LinkWrapper={isGatsby ? LinkWrapper : null} {...p} />
-    ))}
-  </List>
-);
+}) => {
+  const listRef = React.useRef<HTMLUListElement>(null);
+
+  const handleKeyDown: React.KeyboardEventHandler<HTMLUListElement> = (event) => {
+    const list = listRef.current;
+    const { key } = event;
+    const currentFocus = ((list && list.ownerDocument) || document).activeElement;
+
+    if (key === 'ArrowDown') {
+      // Prevent scroll of the page
+      event.preventDefault();
+      moveFocus(list, currentFocus, nextItem);
+    } else if (key === 'ArrowUp') {
+      event.preventDefault();
+      moveFocus(list, currentFocus, previousItem);
+    } else if (key === 'Home') {
+      event.preventDefault();
+      moveFocus(list, null, nextItem);
+    } else if (key === 'End') {
+      event.preventDefault();
+      moveFocus(list, null, previousItem);
+    }
+  };
```

Comment:
Is this going to conflict with @ghengeveld work on the new sidebar?

Reply:
Nop, that's the all point of preventDefaut.
I'll do more testing though

Reply:
escape to close the menu and return focus to button would be nice